packages/appsync-modelgen-plugin/src/utils/process-connections-v2.ts

```diff
@@ -129,14 +129,15 @@ export function processConnectionsV2(
   model: CodeGenModel,
   modelMap: CodeGenModelMap,
   shouldUseModelNameFieldInHasManyAndBelongsTo: boolean = false,
-  isCustomPKEnabled: boolean = false
+  isCustomPKEnabled: boolean = false,
+  shouldUseFieldsInAssociatedWithInHasOne: boolean = false,
 ): CodeGenFieldConnection | undefined {
   const connectionDirective = field.directives.find(d => d.name === 'hasOne' || d.name === 'hasMany' || d.name === 'belongsTo');
 
   if (connectionDirective) {
     switch (connectionDirective.name) {
       case 'hasOne':
-        return processHasOneConnection(field, model, modelMap, connectionDirective, isCustomPKEnabled);
+        return processHasOneConnection(field, model, modelMap, connectionDirective, isCustomPKEnabled, shouldUseFieldsInAssociatedWithInHasOne);
```

Comment:
How do we compute the value for `shouldUseFieldsInAssociatedWithInHasOne`?

Reply:
This value is set in the top level visitors and if not it defaults to `false`. In this PR, we only set this to `true` for Model Introspection visitor which generates the introspection schema.